File: leetcode/js/transform-array-by-parity.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var transformArray = function (nums) {
  for (let i = 0; i < nums.length; i++) {
    nums[i] = nums[i] % 2 === 0 ? 0 : 1;
  }

  return nums.sort((a, b) => a - b);
};

File: leetcode/go/transform-array-by-parity.go
package main

import "sort"

func TransformArray(nums []int) []int {
	for i, v := range nums {
		if v % 2 == 0 {
			nums[i] = 0
		} else {
			nums[i] = 1
		}
	}

	sort.Ints(nums)

	return nums
}